databricks-sdk-java/src/main/java/com/databricks/sdk/core/ApiClient.java

```diff
@@ -29,6 +30,43 @@
  * guessing
  */
 public class ApiClient {
+  static class Builder {
+    private DatabricksConfig config;
+    private Timer timer;
+    private Function<Void, Map<String, String>> authenticateFunc;
+    private Function<Void, String> getHostFunc;
+    private Function<Void, String> getAuthTypeFunc;
+
+    public Builder withDatabricksConfig(DatabricksConfig config) {
+      this.config = config;
+      return this;
+    }
+
+    public Builder withTimer(Timer timer) {
+      this.timer = timer;
+      return this;
+    }
+
+    public Builder withAuthenticateFunc(Function<Void, Map<String, String>> authenticateFunc) {
+      this.authenticateFunc = authenticateFunc;
+      return this;
+    }
+
+    public Builder withGetHostFunc(Function<Void, String> getHostFunc) {
+      this.getHostFunc = getHostFunc;
+      return this;
+    }
+
+    public Builder withGetAuthTypeFunc(Function<Void, String> getAuthTypeFunc) {
+      this.getAuthTypeFunc = getAuthTypeFunc;
+      return this;
+    }
+
+    public ApiClient build() {
+      return new ApiClient(config, timer, authenticateFunc, getHostFunc, getAuthTypeFunc);
```

Comment:
return new ApiClient(this);

Reply:
https://github.com/databricks/databricks-sdk-java/pull/355/commits/652cb0b7c629e2b4a38725ea4069a874352eec88